pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+										// https://help.nextcloud.com/t/nextcloud-office-could-not-establish-connection-to-the-collabora-online-server/185721/8
+										"FOWNER",
+										"SYS_CHROOT",
+										"CHOWN",
+										// https://github.com/chrisingenhaag/helm/blob/main/charts/collabora-code/values.yaml#L64
+										"MKNOD",
+									},
+								},
+								// https://github.com/CollaboraOnline/online/blob/master/docker/from-packages/Dockerfile#L138
+								RunAsUser: ptr.To[int64](100),
+							},
+
+							// Mount certificates
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "collabora-code-tls",
+									MountPath: "/etc/coolwsd/cert.pem",
+									SubPath:   "tls.crt",
+								},
+								{
+									Name:      "collabora-code-tls",
+									MountPath: "/etc/coolwsd/key.pem",
+									SubPath:   "tls.key",
+								},
+								{
+									Name:      "collabora-code-tls",
+									MountPath: "/etc/coolwsd/ca-chain.cert.pem",
+									SubPath:   "ca.crt",
+								},
+								{
+									Name:      "coolwsd-config",
+									MountPath: "/etc/coolwsd/coolwsd.xml",
+									SubPath:   "coolwsd.xml",
+								},
+								{
+									// need to attach it, it's probably generated by coolwsd
+									Name:      "coolwsd-config",
+									MountPath: "/etc/coolwsd/coolkitconfig.xcu",
+									SubPath:   "coolkit.xml",
+								},
+								{
+									// it reports an error if this file is missing
+									// it's necessary to support WOPI protocol: https://learn.microsoft.com/en-us/microsoft-365/cloud-storage-partner-program/rest/
+									Name:      "coolwsd-config",
+									MountPath: "/etc/coolwsd/proof_key",
+									SubPath:   "id_rsa",
+								},
+								{
+									Name:      "tmp",
+									MountPath: "/tmp",
+								},
+								{
+									// it creates chroots there
+									Name:      "child-roots",
+									MountPath: "/opt/cool/child-roots/",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "tmp",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+							Name: "child-roots",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+							Name: "collabora-code-tls",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "collabora-code-tls",
+								},
+							},
+						},
+						{
+							Name: "coolwsd-config",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "collabora-code-coolwsd-config",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if svc.Config.Data["collabora_image"] != "" {
+		sts.Spec.Template.Spec.Containers[0].Image = svc.Config.Data["collabora_image"]
```

Comment:
If we do it like this and the config is empty for some reason, it will downgrade collabora and might break things even worse.

Please put this directly in the definition on line 202. That way if it's empty it will try to create an invalid STS, which will have a much smaller impact.

Reply:
added, please check if I understood correctly

Reply:
> Collabora shouldn't cause any issues even if downgraded, becasue this software never changes any data, it has no database acces or whatsoever, this is why I thought it's ok to have defaults

Sure, but collabora could also have bugs or security issues in this specific version.